drivers/snmp-ups.c

```diff
@@ -1156,7 +1156,20 @@ static struct snmp_pdu **nut_snmp_walk(const char *OID, int max_iteration)
 			snmp_free_pdu(response);
 			break;
 		} else {
-			numerr = 0;
+			/* Checked the "type" field of the returned varbind if it is a type error
+			 * exception (only applicable with SNMPv2 or SNMPv3 protocol) */
```

Comment:
Should it check for the protocol actually used (2/3) here?

Reply:
I think it is not necessary to check the protocol here, it will add a test for nothing.
 Just with snmpv1, this test will never occur.